src/test/java/org/entando/kubernetes/service/EntandoBundleUtilitiesTest.java

```diff
@@ -351,6 +358,62 @@ void receivingAListOfEnvironmentVariableShouldCorrectlyConvertThemToAListOfEnvVa
 
     }
 
+    @Test
+    void testHashCountOnSecretName() {
+        String name = "pn-hasldk12-8dsjahj2-mypluginname-conf";
+        int count = BundleUtilities.countHashesInSecretName(name, 8);
+        assertEquals(2, count);
+
+        name = "pn-hasldk12-8dsjahj2-3dsjahj3-mypluginname-conf";
+        count = BundleUtilities.countHashesInSecretName(name, 8);
+        assertEquals(3, count);
+
+        name = "secretWithNoSeparator";
+        count = BundleUtilities.countHashesInSecretName(name, 8);
+        assertEquals(0, count);
+
+        count = BundleUtilities.countHashesInSecretName(null, 8);
+        assertEquals(0, count);
```

Comment:
just for the future: when you have a list of values to use as test input, you could leverage a map pairing provided input to expected output combined with java streams to save lines of code. Not a big problem here since we are talking about only 10 lines

Reply:
the good ole copy and paste... this is immediate to read, but I understand the point